metadata-ingestion-modules/dagster-plugin/src/datahub_dagster_plugin/sensors/datahub_sensors.py

```diff
@@ -0,0 +1,308 @@
+    jobs: Optional[Sequence[ExecutableDefinition]] = None,
+    default_status: DefaultSensorStatus = DefaultSensorStatus.STOPPED,
+    asset_selection: Optional[CoercibleToAssetSelection] = None,
+    required_resource_keys: Optional[Set[str]] = None,
+) -> SensorDefinition:
+    """Create a sensor on job status change emit lineage to DataHub.
+
+    Args:
+        config (DagsterSourceConfig): DataHub Sensor config
+        name: (Optional[str]): The name of the sensor. Defaults to "datahub_sensor".
+        minimum_interval_seconds: (Optional[int]): The minimum number of seconds that will elapse
+            between sensor evaluations.
+        default_status (DefaultSensorStatus): Whether the sensor starts as running or not. The default
+            status can be overridden from Dagit or via the GraphQL API.
+
+    Examples:
+        .. code-block:: python
+
+            datahub_sensor = make_datahub_sensor(
+               config
+            )
+
+            @repository
+            def my_repo():
+                return [my_job + datahub_sensor]
+
+
+    """
+
+    @sensor(
+        name=name,
+        minimum_interval_seconds=minimum_interval_seconds,
+        description=description,
+        job=job,
+        jobs=jobs,
+        default_status=default_status,
+        asset_selection=asset_selection,
+        required_resource_keys=required_resource_keys,
+    )
+    def datahub_sensor(context):
+        """
+        Sensor which instigate all run status sensors and trigger them based upon run status
+        """
+        for each in DatahubSensors(config).sensors:
+            each.evaluate_tick(context)
+        return SkipReason("Trigger run status sensors if any new runs present...")
+
+    return datahub_sensor
+
+
+class DatahubSensors:
+    def __init__(self, config: Optional[DagsterSourceConfig] = None):
+        """
+        Set dagster source configurations and initialize datahub emitter and dagster run status sensors
+        """
+        if config:
+            self.config = config
+        else:
+            self.config = DagsterSourceConfig()
+
+        self.emitter: DatahubRestEmitter = DatahubRestEmitter(
+            gms_server=self.config.rest_sink_config.server,
+            retry_status_codes=self.config.rest_sink_config.retry_status_codes,
+            retry_max_times=self.config.rest_sink_config.retry_max_times,
+            token=self.config.rest_sink_config.token,
+            connect_timeout_sec=self.config.rest_sink_config.timeout_sec,
+            read_timeout_sec=self.config.rest_sink_config.timeout_sec,
+            extra_headers=self.config.rest_sink_config.extra_headers,
+            ca_certificate_path=self.config.rest_sink_config.ca_certificate_path,
+            client_certificate_path=self.config.rest_sink_config.client_certificate_path,
+            disable_ssl_verification=self.config.rest_sink_config.disable_ssl_verification,
+        )
+        self.emitter.test_connection()
```

Comment:
this will throw if it can't connect - is that what we want?

Reply:
I think it is fine as with this we can see error early on (when sensor is enabled).

Reply:
does this make the entire pipeline fail / not run when datahub is down? if so, I don't think we should be erroring here

Reply:
no, it just fails the sensor run. Sensors doesn't affect actual runs as it only subscribes to an event stream.